spikeextractors/extractors/bindatrecordingextractor/bindatrecordingextractor.py

```diff
@@ -42,12 +42,17 @@ def __init__(self, file_path, sampling_frequency, numchan, dtype, recording_chan
 
         if geom is not None:
             self.set_channel_locations(self._geom)
+
         if 'numpy' in str(dtype):
             dtype_str = str(dtype).replace("<class '", "").replace("'>", "")
             # drop 'numpy
             dtype_str = dtype_str.split('.')[1]
         else:
             dtype_str = str(dtype)
+
+        if gain is not None:
+            self.set_channel_gains(channel_ids=self.get_channel_ids(), gains=gain)
```

Comment:
good, but we should also update the `get_traces` to use the `gain` property instead of the internal `_gain` field